TradeItIosTicketSDK2/TradeItYahooEquityTradingTicketViewController.swift

```diff
@@ -406,41 +407,78 @@ class TradeItYahooTradingTicketViewController: TradeItYahooViewController, UITab
         cell.textLabel?.text = ticketRow.getTitle(forOrder: self.order)
         cell.selectionStyle = .none
 
+        guard let orderCapabilities = self.orderCapabilities else { return cell }
+
         switch ticketRow {
         case .orderAction:
-            cell.detailTextLabel?.text = self.equityOrderCapabilities?.labelFor(field: .actions, value: self.order.action.rawValue)
+            cell.detailTextLabel?.text = orderCapabilities.labelFor(field: .actions, value: self.order.action.rawValue)
         case .quantity:
-            (cell as? TradeItNumericInputCell)?.configure(
-                initialValue: self.order.quantity,
-                placeholderText: "Enter shares",
+            let cell = cell as? TradeItNumericToggleInputCell
+            let quantitySymbol = self.order.quantityType == OrderQuantityType.shares ? "Shares" : "USD"
+            cell?.configure(
                 onValueUpdated: { newValue in
                     self.order.quantity = newValue
-                    self.reload(row: .estimatedCost)
+                    self.reload(row: .estimatedChange)
                     self.setReviewButtonEnablement()
+                },
+                onQuantityTypeToggled: {
+                    let supportedOrderQuantityTypes = orderCapabilities.supportedOrderQuantityTypesFor(action: self.order.action)
+
+                    guard supportedOrderQuantityTypes.count > 0 else { return }
+
+                    let currentIndex = supportedOrderQuantityTypes.index(of: self.order.quantityType ?? supportedOrderQuantityTypes.first ?? .shares) as Int? ?? 0
+                    let nextIndex = (currentIndex + 1) % supportedOrderQuantityTypes.count
+                    let nextOrderQuantityType = supportedOrderQuantityTypes[safe: nextIndex] ?? supportedOrderQuantityTypes.first ?? .shares
+
+                    if self.order.quantityType != nextOrderQuantityType {
+                        self.order.quantityType = nextOrderQuantityType
+
+                        let quantitySymbol = self.order.quantityType == OrderQuantityType.shares ? "Shares" : "USD"
+                        self.order.quantity = nil
+                        cell?.configureQuantityType(
+                            quantitySymbol: quantitySymbol,
+                            quantity: self.order.quantity,
+                            maxDecimalPlaces: orderCapabilities.maxDecimalPlacesFor(orderQuantityType: self.order.quantityType),
+                            showToggle: supportedOrderQuantityTypes.count > 1
+                        )
+                    }
                 }
             )
+            let supportedOrderQuantityTypes = orderCapabilities.supportedOrderQuantityTypesFor(action: self.order.action)
```

Comment:
Looks like the copy-pasta for `quantitySymbol` appears before the `cell?.configure(` call on L416 but the copy-pasta for this var (`supportedOrderQuantityTypes`) is coming after. Pretty tough to digest

Reply:
Yeah I'll spend some time trying to tighten this up before merging.